admin/app/components/solidus_admin/orders/address/component.html.erb

```diff
@@ -0,0 +1,70 @@
+<div class="<%= stimulus_id %>">
+  <%= page do %>
+    <%= page_header do %>
+      <%= page_header_back(solidus_admin.edit_order_path(@order)) %>
+      <%= page_header_title(t(".title", type: t(".#{address_type_action}_address"))) %>
+      <%= page_header_actions do %>
+        <%= render component("ui/button").new(
+          tag: :a,
+          scheme: :secondary,
+          text: t(".cancel"),
+          href: solidus_admin.edit_order_path(@order)
+        ) %>
+
+        <%= render component("ui/button").new(
+          tag: :button,
+          text: t(".save"),
+          form: form_id
+        ) %>
+      <% end %>
+    <% end %>
+
+    <%= page_with_sidebar do %>
+      <%= page_with_sidebar_main do %>
+        <%= render component('ui/panel').new do %>
+          <%= form_for @order, url: solidus_admin.send("order_#{@type}_address_path", @order), html: { id: form_id } do |form| %>
+            <div class="w-full flex flex-col mb-4">
+              <h2 class="text-sm mb-4 font-semibold"><%= t(".#{address_type_action}_address") %></h2>
+              <div class="w-full flex gap-4">
+                <%= form.fields_for :"#{@type}_address" do |address_form| %>
+                  <%= render component('ui/forms/address').new(form: address_form, disabled: false) %>
+                <% end %>
+              </div>
+
+              <label class="flex gap-2 items-center">
+                <%= form.hidden_field "use_#{address_type_action}", value: '0', id: false %>
+
+                <%= render component("ui/forms/checkbox").new(
+                  name: "#{form.object_name}[use_#{address_type_action}]",
+                  checked: form.object.send("#{@type}_address").new_record? && form.object.bill_address == form.object.ship_address,
+                  value: '1'
+                ) %>
+
+                <span class="body-text-sm">
+                  <%= t(".use_this_address", type: "#{@type == 'ship' ? 'billing' : 'shipping'}") %>
```

Comment:
nit: if the translations supports directly `ship` and `bill` this line becomes a lot simpler:

```yaml
use_this_address:
  bill: "Use this address also for Shipping"
  ship: "Use this address also for Billing"
```